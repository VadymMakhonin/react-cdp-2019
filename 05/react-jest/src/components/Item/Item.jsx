import React from 'react';
import { dateFormater } from './../../utils/common';


export const Item = ({ todo, onRemove, onChange }) => (
    <div className="row" key={todo.id}>
        <div className="col-1">
            <input type="checkbox" checked={todo.complited} onChange={(event) => {
                onChange(todo.id, event.target.checked);
            }} />
        </div>
        <div className="col-1">{todo.id}</div>
        <div className="col-7">{todo.title}</div>
        <div className="col-1">{dateFormater(todo.created)}</div>
        <div className="col-2 text-right">
            <button className="btn btn-danger" onClick={() => { onRemove(todo.id) }}>
                Remove
                </button>
        </div>

    </div>
);