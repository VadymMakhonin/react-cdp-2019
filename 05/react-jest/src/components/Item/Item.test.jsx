import React from 'react';
import {Item} from './Item';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';


describe('Item component', () => {

    let todo;

    beforeAll(() => {
        Date.now = jest.fn(() => Date.UTC(2017, 0, 1, 1, 1, 1, 1));

        todo = {
            title: 'test',
            id: 1,
            complited: false,
            created: Date.now()
        }
    });

    it('should be render correctly', () => {
        const component = renderer.create(<item todo={todo} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should be call onRemove', () => {
        const onRemove = jest.fn();
        const component = mount(<Item todo={todo} onRemove={onRemove}/>);
        const btn = component.find('.btn').simulate('click');
        expect(onRemove).toHaveBeenCalled();
    });
})