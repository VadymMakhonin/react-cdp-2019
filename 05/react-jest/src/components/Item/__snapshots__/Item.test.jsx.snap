// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should be render correctly 1`] = `
<item
  todo={
    Object {
      "complited": false,
      "created": 1483232461001,
      "id": 3,
      "title": "test",
    }
  }
/>
`;
